GVFS/GVFS.Platform.Mac/MacPlatform.cs

```diff
@@ -140,5 +141,41 @@ public override bool IsGitStatusCacheSupported()
         {
             return new MacFileBasedLock(fileSystem, tracer, lockPath);
         }
+
+        public override bool OpenAndReadFile(string fileName)
+        {
+            return NativeFileReader.OpenAndReadFile(fileName);
+        }
+
+        private class NativeFileReader
+        {
+            // #define O_RDONLY     0x0000      
+            public const int ReadOnly = 0x0000;
+
+            public static bool OpenAndReadFile(string fileName)
+            {
+                int fileDescriptor = Open(fileName, ReadOnly);
+                int success = ReadOneByte(fileDescriptor);
+
+                if (success == -1)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            private static unsafe int ReadOneByte(int fileDescriptor)
+            {
+                byte[] buffer = new byte[1];
+                return Read(fileDescriptor, buffer, 1);
+            }
+
+            [DllImport("libc", EntryPoint = "open", SetLastError = true)]
+            private static extern int Open(string path, int flag);

```

Comment:
Not something for this PR, but at some point we might want to move all the native interop calls to a common class in `MacPlatform`

Reply:
I agree, I was thinking the same thing while working on this. Want me to open a work item for it?